src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -35,20 +50,94 @@ public LibGpiodDriver()
             _pinNumberToSafeLineHandle = new Dictionary<int, SafeLineHandle>(PinCount);
         }
 
+        private SafeLineHandle SubscribeForEvent(int pinNumber, PinEventTypes eventType)
+        {
+            int eventSuccess = -1; 
+            SafeLineHandle pinHandle = Interop.GetChipLineByOffset(_chip, pinNumber);
+            if (pinHandle != null)
+            {
+                if (eventType.HasFlag(PinEventTypes.Rising) || eventType.HasFlag(PinEventTypes.Falling))
+                {
+                    eventSuccess = Interop.RequestBothEdgesEventForLine(pinHandle, $"Listen {pinNumber} for both edge event");
+                }
+                else
+                {
+                    throw ExceptionHelper.GetArgumentException(ExceptionResource.InvalidEventType);
+                }
+            }
+
+            if (eventSuccess < 0)
+            {
+                throw ExceptionHelper.GetIOException(ExceptionResource.RequestEventError, pinNumber, Marshal.GetLastWin32Error());
+            }
+
+            return pinHandle;
+        }
+
+        private Task InitializeEventDetectionTask(LibGpiodDriverEventHandler eventHandler, CancellationToken token)
+        {
+            return Task.Run(() =>
+            {       
+                while (!(token.IsCancellationRequested || _disposing))
```

Comment:
alternative: `!token.IsCancellationRequested && !_disposing`

Reply:
which one is better you think? for me mine better 😛

Reply:
I'm fine either way with slight preference toward mine :)